modules/tinymce/src/core/test/ts/browser/EditorUploadTest.ts

```diff
@@ -25,174 +23,167 @@ UnitTest.asynctest('browser.tinymce.core.EditorUploadTest', (success, failure) =
-    return Arr.bind(steps, function (step) {
-      return [ step, teardown(editor) ];
-    });
-  };
+  const appendTeardown = (editor: Editor, steps: Step<any, any>[]) =>
+    Arr.bind(steps, (step) =>
+      [ step, teardown(editor) ]
+    );
 
-  const imageHtml = function (uri: string) {
-    return DOMUtils.DOM.createHTML('img', { src: uri });
-  };
+  const imageHtml = (uri: string) =>
+    DOMUtils.DOM.createHTML('img', { src: uri });
 
-  const assertResult = function (editor: Editor, uploadUri: string, uploadedBlobInfo: BlobInfo, result: UploadResult[]) {
+  const assertResult = (editor: Editor, title: string, uploadUri: string, uploadedBlobInfo: BlobInfo, result: UploadResult[]) => {
     const firstResult = result[0];
-    LegacyUnit.strictEqual(result.length, 1);
-    LegacyUnit.strictEqual(firstResult.status, true);
-    LegacyUnit.strictEqual(firstResult.element.src.indexOf(uploadedBlobInfo.id() + '.png') !== -1, true);
-    LegacyUnit.strictEqual(firstResult.uploadUri, uploadUri);
-    LegacyUnit.strictEqual(firstResult.blobInfo.id(), uploadedBlobInfo.id());
-    LegacyUnit.strictEqual(firstResult.blobInfo.name(), uploadedBlobInfo.name());
-    LegacyUnit.strictEqual(firstResult.blobInfo.filename(), uploadedBlobInfo.filename());
-    LegacyUnit.strictEqual(firstResult.blobInfo.blob(), uploadedBlobInfo.blob());
-    LegacyUnit.strictEqual(firstResult.blobInfo.base64(), uploadedBlobInfo.base64());
-    LegacyUnit.strictEqual(firstResult.blobInfo.blobUri(), uploadedBlobInfo.blobUri());
-    LegacyUnit.strictEqual(firstResult.blobInfo.uri(), uploadedBlobInfo.uri());
-    LegacyUnit.equal('<p><img src="' + uploadedBlobInfo.filename() + '" /></p>', editor.getContent());
+    Assert.eq(title, 1, result.length);
+    Assert.eq(title, true, firstResult.status);
+    Assert.eq(title, true, firstResult.element.src.indexOf(uploadedBlobInfo.id() + '.png') !== -1);
+    Assert.eq(title, uploadUri, firstResult.uploadUri);
+    Assert.eq(title, uploadedBlobInfo.id(), firstResult.blobInfo.id());
+    Assert.eq(title, uploadedBlobInfo.name(), firstResult.blobInfo.name());
+    Assert.eq(title, uploadedBlobInfo.filename(), firstResult.blobInfo.filename());
+    Assert.eq(title, uploadedBlobInfo.blob(), firstResult.blobInfo.blob());
+    Assert.eq(title, uploadedBlobInfo.base64(), firstResult.blobInfo.base64());
+    Assert.eq(title, uploadedBlobInfo.blobUri(), firstResult.blobInfo.blobUri());
+    Assert.eq(title, uploadedBlobInfo.uri(), firstResult.blobInfo.uri());
+    Assert.eq(title, editor.getContent(), '<p><img src="' + uploadedBlobInfo.filename() + '" /></p>');
 
     return result;
   };
 
-  const hasBlobAsSource = function (elm) {
-    return elm.src.indexOf('blob:') === 0;
-  };
+  const hasBlobAsSource = (elm) =>
+    elm.src.indexOf('blob:') === 0;
 
-  suite.asyncTest('_scanForImages', function (editor, done, die) {
+  const sScanForImages = (editor) => Log.step('TBA', '_scanForImages', Step.async((done, die) => {
```

Comment:
Types for `editor` here please and all the place below (this wasn't needed previously as it was inferred from `suite.asyncTest`)